spec/models/manageiq/providers/hawkular/inventory/parser/middleware_manager_spec.rb

```diff
@@ -97,79 +97,149 @@ def inventory_object_data(inventory_object)
     end
   end
 
-  describe 'parse_availability' do
-    it 'handles simple data' do
-      resources_by_metric_id = {
-        'resource_id_1' => [{}],
-        'resource_id_2' => [{}, {}],
-        'resource_id_3' => [{}],
-        'resource_id_4' => [{}]
-      }
-      availabilities = [
-        {
-          'id'   => 'resource_id_1',
-          'data' => [{ 'value' => 'up' }]
-        },
-        {
-          'id'   => 'resource_id_2',
-          'data' => [{ 'value' => 'down' }]
-        },
-        {
-          'id'   => 'resource_id_3',
-          'data' => [{ 'value' => 'something_else' }]
-        },
-        {
-          'id'   => 'resource_id_4',
-          'data' => []
-        }
-      ]
-
-      parsed_resources_with_availability = {
-        'resource_id_1' => [{
-          :status => 'Enabled'
-        }],
-        'resource_id_2' => [
-          {
-            :status => 'Disabled'
-          },
-          {
-            :status => 'Disabled'
-          }
-        ],
-        'resource_id_3' => [{
-          :status => 'Unknown'
-        }],
-        'resource_id_4' => [{
-          :status => 'Unknown'
-        }]
-      }
-      expect(parser.parse_availability(availabilities, resources_by_metric_id))
-        .to eq(parsed_resources_with_availability)
+  describe 'fetch_deployment_availabilities' do
+    it 'resolves availabilities correctly' do
+      parser = described_class.new
+
+      # Define a mocked persister.
+      # It has two feeds:
+      # - Feed 1 has two deployments with ids: d1, d2
+      # - Feed 2 has three deployments with ids: d1, d2, d3, d4
+      parser.persister = double("Mocked persister")
+      deployments = []
+      deployments << OpenStruct.new(:manager_uuid => '/t;hawkular/f;1/r;s1/r;d1')
+      deployments << OpenStruct.new(:manager_uuid => '/t;hawkular/f;1/r;s1/r;d2')
+      deployments << OpenStruct.new(:manager_uuid => '/t;hawkular/f;2/r;s1/r;d1')
+      deployments << OpenStruct.new(:manager_uuid => '/t;hawkular/f;2/r;s1/r;d2')
+      deployments << OpenStruct.new(:manager_uuid => '/t;hawkular/f;2/r;s1/r;d3')
+      deployments << OpenStruct.new(:manager_uuid => '/t;hawkular/f;2/r;s1/r;d4')
+      allow(parser.persister).to receive(:middleware_deployments) { deployments }
+
+      # Define a mocked collector.
+      # It will return five metrics for deployments. Relationships metric_id => deployment_ids are:
+      # - m1 => feed 1, d1
+      # - m2 => feed 1, d2; feed 2, d3
+      # - m3 => feed 2, d1
+      # - m4 => feed 2, d2
+      # - m5 => feed 2, d4
+      #
+      # For metric 'm5', no metric data will be returned (simulating missing metric).
+      parser.collector = double("Mocked collector")
+      allow(parser.collector).to receive(:metrics_for_metric_type) do |feed, metric_type_id|
```

Comment:
> First I'd change so every test case tests one of the possibilities.

I agree! This test is testing a lot. I will break it into the very specific cases.

> Another option is to extract the mock behaviour into its own class, and also test that class.

There are some cases where I still will need to add if/else logic to the mock. But wait until I broke this test into the specific test cases to see if the mock is acceptable. If not acceptable, then I'll place it separately.